### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by gab3000**
at checkpoint#3

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -173,143 +173,143 @@ struct CarWash //2)
...
 /*
-6)
+6)DA converters
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1)amount of converters
```
Comment (matkatmusic) [inline main.cpp]: I think you mean `number of channels`
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -173,143 +173,143 @@ struct CarWash //2)
...
 /*
-7)
+7)AD converters
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1)amount of converters
```
Comment (matkatmusic) [inline main.cpp]: I think you mean `number of channels`
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -173,143 +173,143 @@ struct CarWash //2)
...
-    1)
-    2)
-    3)
-    4)
-    5)
+    1)size
+    2)location on the board
+    3)color
+    4)material
+    5)useful life
```
Comment (matkatmusic) [inline main.cpp]: `useful life` -> `number of clicks before failure`?

**Review by matkatmusic — APPROVED** [review]
added some suggestions.   You can code them up as the instructions indicate.

File: main.cpp (hunk 1 above)
Comment (gab3000) [reply main.cpp]: ok I use number of channels instead of amount of converters

File: main.cpp (hunk 2 above)
Comment (gab3000) [reply main.cpp]: ok I use number of channels instead of amount of converters

File: main.cpp (hunk 3 above)
Comment (gab3000) [reply main.cpp]: yes, I agree

File: main.cpp [hunk 4]
```diff
@@ -283,35 +445,98 @@ struct CarWash //2)
...
+    //1)size
+    int memorySize = 1000;
+    //2)type of memory (ram , eprom, rom, flash)
+    char memoryType = 'r';
+    //3)clock speed
+    int memorySpeed = 1000;
+    //4)slot for flash card
+    char memoryFlash = 'y';
+    //5)power consumption
+    int powerComsumption = 200;
```
Comment (matkatmusic) [inline main.cpp]: this class is called `Memory`.  
Imagine an instance being created:
```
Memory memory;
```
now you want to access the memory size and speed:
```
std::cout << memory.memorySize << "\n";
std::cout << memory.memorySpeed << "\n";
```
using the word `memory` in your member variable names for a class named `Memory` is redundant
- [x] revise.

File: main.cpp [hunk 5]
```diff
@@ -283,35 +445,98 @@ struct CarWash //2)
...
+    //1)size
+    float buttonHeight = 0.05f;
+    //2)location on the board
+    char buttonLocation = 't';
+    //3)color
+    char buttonColor = 'r';
+    //4)material
+    char buttonMaterial = 'p';
+    //5)number of clicks before failure
+    int buttonClicksBeforeFailure = 10000;
```
Comment (matkatmusic) [inline main.cpp]: using 'button' in the member variable names of a class named `Button` is redundant
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -241,35 +344,94 @@ struct CarWash //2)
...
+    //1)speed in Mb per second
+    int usbSpeed = 40;
+    //2)standard (1.0 or 2.0 for USB ) 
+    int usbStandard = 1;
+    //3)cable length
+    float cableLength = 5; 
+    //4)connector type
+    char usbConnectorType = 'A';
+    //5)current it can receive for powering the board
+    float currentDrawForPower = 50;
```
Comment (matkatmusic) [inline main.cpp]: using 'usb' in the member variable names of a class named `UsbInterface` is redundant
- [x] revise

File: main.cpp [hunk 7]
```diff
@@ -199,7 +231,39 @@ struct CarWash //2)
     2)power devices trough USB
     3)display the battery charge level
  */
+struct CarJumpstarter
```
Comment (matkatmusic) [inline main.cpp]: most jump starters are not limited to cars only, but to all vehicle types. 
- [x] rename to `JumpStarter`

**Review by matkatmusic — CHANGES_REQUESTED** [review]
You've got some redundancy that needs addressing.

File: main.cpp (hunk 4 above)
Comment (gab3000) [reply main.cpp]: ok, removed word memory from member variables.

File: main.cpp (hunk 6 above)
Comment (gab3000) [reply main.cpp]: ok, removed word 'usb' from member variables.

File: main.cpp (hunk 5 above)
Comment (gab3000) [reply main.cpp]: ok, removed word 'button' from member variables.

File: main.cpp (hunk 7 above)
Comment (gab3000) [reply main.cpp]: ok, renamed CarJumpstarter to JumpStarter.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**